beacon-chain/state/v1/beacon_state.go

```diff
@@ -0,0 +1,50 @@
+// +build !minimal
+
+package v1
+
+import (
+	"sync"
+
+	eth2types "github.com/prysmaticlabs/eth2-types"
+	"github.com/prysmaticlabs/go-bitfield"
+	customtypes "github.com/prysmaticlabs/prysm/beacon-chain/state/custom-types"
+	"github.com/prysmaticlabs/prysm/beacon-chain/state/fieldtrie"
+	"github.com/prysmaticlabs/prysm/beacon-chain/state/stateutil"
+	"github.com/prysmaticlabs/prysm/beacon-chain/state/types"
+	ethpb "github.com/prysmaticlabs/prysm/proto/prysm/v1alpha1"
+)
+
+// BeaconState defines a struct containing utilities for the Ethereum Beacon Chain state, defining
+// getters and setters for its respective values and helpful functions such as HashTreeRoot().
+type BeaconState struct {
+	genesisTime                 uint64                      `ssz-gen:"true"`
+	genesisValidatorsRoot       customtypes.Byte32          `ssz-gen:"true" ssz-size:"32"`
+	slot                        eth2types.Slot              `ssz-gen:"true"`
+	fork                        *ethpb.Fork                 `ssz-gen:"true"`
+	latestBlockHeader           *ethpb.BeaconBlockHeader    `ssz-gen:"true"`
+	blockRoots                  *customtypes.StateRoots     `ssz-gen:"true" ssz-size:"8192,32"`
```

Comment:
Why use state roots for block roots

Reply:
I created only one custom type for both and called it `StateRoots`. This is a pretty unfortunate name and I will change it to `BeaconRoots` in the upcoming cleanup PR. Alternatively I can create another custom type called `BlockRoots`. Which one would you prefer?

Reply:
I think we should be separating individual fields here rather than specifying the same custom type, the latter is preferrable